src/snowflake/connector/network.py

```diff
@@ -994,11 +1006,8 @@ def _request_exec(
         logger.debug("socket timeout: %s", socket_timeout)
         try:
             if not catch_okta_unauthorized_error and data and len(data) > 0:
-                gzdata = BytesIO()
-                gzip.GzipFile(fileobj=gzdata, mode="wb").write(data.encode("utf-8"))
-                gzdata.seek(0, 0)
                 headers["Content-Encoding"] = "gzip"
-                input_data = gzdata
+                input_data = gzip.compress(data.encode("utf-8"))
```

Comment:
code change itself looks good to me.

have you sanity checked that the compress mechanisms under two approaches are the same?
you know, from the issue description looks like closing the gzip file is also an option. but if the effect is the same, then less code wins.

Reply:
Yes, I have manually compressed a file with both methods and then uncompressed the created GZIP bytes and got back the same things.
The effect should be the same.